src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs

```diff
@@ -2,37 +2,129 @@
+            {
+                StringBuilder stringBuilder = new();
+
+                comboBox.EventsCount = 0;
+                int keyPressesCount = 9;
+                for (int i = 0; i < keyPressesCount; i++)
+                {
+                    await Task.Delay(100);
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard
+                        .Sleep(100)
+                        .KeyPress(WindowsInput.Native.VirtualKeyCode.UP));
+
+                    await Task.Delay(100);
+
+                    stringBuilder.Append($"i:{i + 1}, eventsCount:{comboBox.EventsCount}\n");
+                }
+
+                TestOutputHelper.WriteLine($"{stringBuilder}");
+                Assert.Equal(keyPressesCount, comboBox.EventsCount);
+            },
+            numberOfItemsToAdd: 10,
+            selectedIndex: 9);
+        }
+
+        [WinFormsFact]
+        public async Task ComboBox_Select_Item_By_DownArrowKeyAsync()
+        {
+            await RunComboBoxTestAsync(async (form, comboBox) =>
+            {
+                comboBox.EventsCount = 0;
+                int keyPressesCount = 9;
+                for (int i = 0; i < keyPressesCount; i++)
+                {
+                    await InputSimulator.SendAsync(
+                        form,
+                        inputSimulator => inputSimulator.Keyboard.KeyPress(WindowsInput.Native.VirtualKeyCode.DOWN));
+
+                    Assert.Equal(i + 1, comboBox.EventsCount);
+                }
+
+                Assert.Equal(keyPressesCount, comboBox.EventsCount);
+            },
+            numberOfItemsToAdd: 10,
+            selectedIndex: 0);
+        }
+
+        private async Task RunComboBoxTestAsync(Func<Form, ComboBoxWithSelectCounter, Task> runTest, int numberOfItemsToAdd, int selectedIndex)
+        {
+            await RunSingleControlTestAsync(
+                testDriverAsync: runTest,
+                createControl: () =>
+                {
+                    ComboBoxWithSelectCounter control = new();
+                    AddItems(control, numberOfItemsToAdd);
+                    control.SelectedIndex = selectedIndex;
+                    return control;
+                },
+                createForm: () =>
+                {
+                    return new()
+                    {
+                        Size = new(500, 300),
+                    };
+                });
+
+            static void AddItems(ComboBoxWithSelectCounter comboBox, int numToAdd)
+            {
+                for (int i = 0; i < numToAdd; i++)
+                {
+                    comboBox.Items.Add($"Item {i}");
+                }
+            }
+        }
+
+        private class ComboBoxWithSelectCounter : ComboBox
+        {
+            public int EventsCount;
+            protected override void OnSelectedIndexChanged(EventArgs e)
```

Comment:
```suggestion
            public int EventsCount { get; private set; }

            protected override void OnSelectedIndexChanged(EventArgs e)
```

Lines:47,75 can go since the property is set to 0 when the control is created.

Reply:
> Lines:47,75 can go since the property is set to 0 when the control is created.

We need to reset the EventCouns property to 0 at 45, 45 because at RunComboBoxTestAsync() this property rises to 1 since control is being selected once right after creation.
So at the very start of ComboBox_Select_Item_By_UpArrowKeyAsync and ComboBox_Select_Item_By_DownArrowKeyAsync the EventsCount property == 1.
But we need to count only selections caused by presses of arrow keys.

I can create some kind of ResetCount() method, but it may be redundant.

Reply:
Please add a comment to the tests explaining this. Thanks